src/components/TaskInput.tsx

```diff
@@ -0,0 +1,74 @@
+import { useState, useCallback, ChangeEventHandler } from 'react';
+import { useDispatch } from 'react-redux';
+import { TextField } from '@mui/material';
+import { styled } from '@mui/material/styles';
+import AddBoxIcon from '@mui/icons-material/AddBox';
+import { addTodo } from '../actions/actionCreators';
+import { getNewTaskObject } from '../utils/getNewTaskObject';
+import { AddNewTodoModal } from './AddNewTodoModal';
+
+export default function TaskInput() {
+  const [isModalOpen, setModalOpen] = useState(false);
+  const [taskValue, setTaskValue] = useState('');
+  const dispatch = useDispatch();
+
+  const handleModalOpen = useCallback(() => {
```

Comment:
You can create one function and pass true/false, or make setModalOpen(!isModalOpen) in function